backend/src/main/java/com/google/sps/FindMatchQuery.java

```diff
@@ -50,30 +54,96 @@ public Match findMatch(Participant newParticipant) {
     List<Participant> sameDurationParticipants =
         participantDatastore.getParticipantsWithDuration(duration);
 
-    // Set reference date time using clock
-    long currentTimeMillis = clock.millis();
-
-    long newEndTimeAvailable = newParticipant.getEndTimeAvailable();
+    long firstEndTimeAvailable = newParticipant.getEndTimeAvailable();
+    MatchPreference firstMatchPreference = newParticipant.getMatchPreference();
+    String firstRole = newParticipant.getRole();
+    String firstProductArea = newParticipant.getProductArea();
 
     // Compare new participant preferences with other participants to find match
     for (Participant currParticipant : sameDurationParticipants) {
       // Check if participants are both free for that duration + extra
-      long currEndTimeAvailable = currParticipant.getEndTimeAvailable();
-      long earliestEndTimeAvailable = Math.min(newEndTimeAvailable, currEndTimeAvailable);
-      boolean compatibleTime =
-          (currentTimeMillis + TimeUnit.MINUTES.toMillis(duration + PADDING_MINUTES))
-              < earliestEndTimeAvailable;
-
-      if (compatibleTime) {
-        // Create and return match
-        return new Match(
-            newParticipant.getUsername(),
-            currParticipant.getUsername(),
-            duration,
-            currentTimeMillis);
+      long secondEndTimeAvailable = currParticipant.getEndTimeAvailable();
+      if (!isCompatibleTime(duration, firstEndTimeAvailable, secondEndTimeAvailable)) {
```

Comment:
Can you add a TODO + issue to push this into the datastore query? We should be able to query for all users with a matching duration %% whose endTimeAvailable is at least duration + PADDING minutes after now

Reply:
I changed it in this PR, but thanks for pointing this out. This was one of the reasons why we chose to only allow same durations to be matched.